android/src/main/java/com/revenuecat/purchases/hybridcommon/common.kt

```diff
@@ -184,6 +183,72 @@ fun purchasePackage(
     }
 }
 
+fun purchaseSubscriptionOption(
+    activity: Activity?,
+    productIdentifier: String,
+    optionIdentifier: String,
+    googleOldProductId: String?,
+    googleProrationMode: GoogleProrationMode?,
+    googleIsPersonalizedPrice: Boolean?,
+    onResult: OnResult
+) {
+    if (activity != null) {
+        val onReceiveStoreProducts: (List<StoreProduct>) -> Unit = { storeProducts ->
+            val optionToBuy = storeProducts.mapNotNull { storeProduct ->
+                storeProduct.subscriptionOptions?.map { Pair(storeProduct, it) }
+            }.flatten().firstOrNull { (storeProduct, subscriptionOption) ->
+                storeProduct.purchasingData.productId == productIdentifier && subscriptionOption.id == optionIdentifier
+            }?.second
+
+            if (optionToBuy != null) {
+                val purchaseParams = PurchaseParams.Builder(optionToBuy, activity)
```

Comment:
Hmm this is a bit of a nitpick and actually belongs in the android sdk, but I would have preferred activity to be the first parameter here...